build-logic/convention/build.gradle.kts

```diff
@@ -14,15 +14,23 @@
  * limitations under the License.
  */
 
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
 plugins {
     `kotlin-dsl`
 }
 
 group = "com.google.samples.apps.nowinandroid.buildlogic"
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_11
-    targetCompatibility = JavaVersion.VERSION_11
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
```

Comment:
I understand that AGP now requires JDK 17 to run, but why would you bump the source/targetCompatibility, and jvmTarget to 17? is this required as well?

Reply:
Basically to have the latest versions, toolchains and formats. We talk about importance of updating early and often - https://www.droidcon.com/2022/08/03/5-ways-to-improve-your-android-build-productivity/

Reply:
Thanks about your answer. I dont doubt the fact is important to update early and often. Using the latest java toolchain makes sense, but I cant find resource to understand the benefits on bumping the source compatibility and kotlin jvm target here. I assume that the generated bytecode will also need to be desugared before to be dexed? 
are there any references to documentation about this part?

Reply:
technically you still get compiler improvements depending on what your actually using. you can still use JDK19 and set targetCompatibility to 17

Reply:
In the interest of getting this change merged I will resolve this conversation. 

@lub0s thanks for your suggestion makes a lot of sense to me. I'll pass this on to the tools team for prioritisation.